modules/http/src/main/java/org/mule/module/http/internal/request/grizzly/GrizzlyHttpClient.java

```diff
@@ -515,4 +480,109 @@ public void stop()
     {
         asyncHttpClient.close();
     }
+
+    private class BodyDeferringWorkManagerSource implements AsyncHandler<Response>, WorkManagerSource
```

Comment:
Just because I was replacing "WorkManagerSourceAsyncCompletionHandler" and it was both a handler and a WorkManagerSource but I guess a better name would be "WorkManagerSourceBodyDeferringHandler" or something like that.